tasks/keyInfraSetters.ts

```diff
@@ -35,6 +36,40 @@ task("rotate:IntermediateKey")
     }
   });
 
+task("hd:rotate:IntermediateKey")
+  .addParam("verifiercontract", "Address of the Access Token Verifier Contract")
+  .addParam("newintermediate", "Address of the new intermediate key")
+  .setAction(async function (taskArguments: TaskArguments, { ethers }) {
+    const newIntermediateKey = taskArguments.newintermediate;
+    if (!newIntermediateKey || newIntermediateKey === ethers.constants.AddressZero) {
+      throw new Error("Invalid new intermediate key");
+    }
+
+    const ledger = new LedgerSigner(ethers.provider);
+
+    const accessTokenVerifier = <AccessTokenVerifier>(
+      await ethers.getContractAt("AccessTokenVerifier", taskArguments.verifiercontract)
+    );
+
+    let tx: ContractTransaction | null = null;
```

Comment:
Interesting notation, never saw it before, could you clarify it for my curiosity? :)

Reply:
Maybe stupid but just very explicit declaration here.

Using the `|` notation as type union to declare the potential types of `tx` as both `ContractTransaction` and `null` and then initialising it to `null`.